src/main/kotlin/no/nav/familie/tilbake/dokumentbestilling/felles/EksterneDataForBrevService.kt

```diff
@@ -40,10 +49,93 @@ class EksterneDataForBrevService(
         personinfo: Personinfo,
         brevmottager: Brevmottager,
         verge: Verge?,
-        fagsystem: Fagsystem
+        fagsystem: Fagsystem,
+        behandlingId: UUID
+    ): Adresseinfo {
+        if (featureToggleService.isEnabled(FeatureToggleConfig.DSITRIBUER_TIL_MANUELLE_BREVMOTTAKERE)) {
+            val manuelleBrevmottakere = manuellBrevmottakerService.hentBrevmottakere(behandlingId)
+            if (manuelleBrevmottakere.isNotEmpty()) {
+                val (manuellVerge, manuellBruker) = manuelleBrevmottakere
+                    .partition { it.type == MottakerType.VERGE }
+                    .let { it.first.singleOrNull() to it.second.singleOrNull() }
+
+                val brukerNavn = manuellBruker?.navn ?: personinfo.navn
+                val vergeNavn: String? = hentVergeNavn(manuellVerge, verge)
+
+                if (brevmottager == Brevmottager.VERGE) {
+                    if (manuellVerge == null && verge == null) {
+                        throw Feil("Brevmottager for behandling $behandlingId was ${Brevmottager.VERGE} but no aktivVerge was found")
+                    }
+                    return hentAdresseForVerge(manuellVerge, vergeNavn, brukerNavn, verge)
+                } else {
+                    return hentAdresseForBruker(manuellBruker, brukerNavn, vergeNavn, personinfo)
+                }
+            } else {
+                return verge?.let {
+                    hentAdresse(
+                        vergeType = it.type,
+                        organisasjonsnummer = it.orgNr,
+                        vergeNavn = it.navn,
+                        personinfo = personinfo,
+                        brevmottager = brevmottager,
+                        vergeIdent = it.ident,
+                        fagsystem = fagsystem
+                    )
+                }
+                    ?: hentAdresse(personinfo)
+            }
+        } else {
+            return verge?.let {
+                hentAdresse(
+                    vergeType = it.type,
+                    organisasjonsnummer = it.orgNr,
+                    vergeNavn = it.navn,
+                    personinfo = personinfo,
+                    brevmottager = brevmottager,
+                    vergeIdent = it.ident,
+                    fagsystem = fagsystem
+                )
+            }
+                ?: hentAdresse(personinfo)
+        }
+    }
```

Comment:
Koden kunne vært forenklet litt her ved å droppe else-branch'ene, tror jeg. siden if'en ender med `return` og else'ene kjører samme kode..
```suggestion
                }
            }
        }
        
        return verge?.let {
            hentAdresse(
                vergeType = it.type,
                organisasjonsnummer = it.orgNr,
                vergeNavn = it.navn,
                personinfo = personinfo,
                brevmottager = brevmottager,
                vergeIdent = it.ident,
                fagsystem = fagsystem
            )
        }
            ?: hentAdresse(personinfo)
    }
```

Reply:
Enig, hadde veldig mindsett på hvordan det kom til å se ut med og uten toggle, men blir finere slik :)